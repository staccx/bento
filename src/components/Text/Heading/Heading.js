import React from "react"
import PropTypes from "prop-types"
import styled, { css } from "styled-components"
import {
  font,
  fontFamily,
  applyVariants,
  commonPropTypes
} from "../../../theming"
import themeProps from "./Headling.themeProps"
import { componentCreateFactory } from "../../../theming/utils/createVariantsFunctionFactory"

/**
 * Heading is Bentos header tag, it take a prop called Level with a range from 1-6.
 * This level tag is what sets the size of the heading.
 */
const Heading = ({ level, children, className, ...restProps }) => {
  switch (level) {
    case "h1":
    case "H1":
    case "1":
    case 1:
      return (
        <HeadingH1 level={1} className={className} {...restProps}>
          {children}
        </HeadingH1>
      )

    case "h2":
    case "H2":
    case "2":
    case 2:
      return (
        <HeadingH2 level={2} className={className} {...restProps}>
          {children}
        </HeadingH2>
      )

    case "h3":
    case "H3":
    case "3":
    case 3:
      return (
        <HeadingH3 level={3} className={className} {...restProps}>
          {children}
        </HeadingH3>
      )

    case "h4":
    case "H4":
    case "4":
    case 4:
      return (
        <HeadingH4 level={4} className={className} {...restProps}>
          {children}
        </HeadingH4>
      )

    case "h5":
    case "H5":
    case "5":
    case 5:
      return (
        <HeadingH5 level={5} className={className} {...restProps}>
          {children}
        </HeadingH5>
      )

    case "h6":
    case "H6":
    case "6":
    case 6:
      return (
        <HeadingH6 level={6} className={className} {...restProps}>
          {children}
        </HeadingH6>
      )

    default:
      return (
        <HeadingH1 level={1} className={className} {...restProps}>
          {children}
        </HeadingH1>
      ) // ðŸ‘ˆ we always have to return something
  }
}

export const HeadingsStyle = css`
  font-family: ${fontFamily.heading};
  ${applyVariants(themeProps.heading)};
`

const HeadingH1 = styled.h1`
  font-size: ${font.h1};
  ${HeadingsStyle};
`

const HeadingH2 = styled.h2`
  font-size: ${font.h2};
  ${HeadingsStyle};
`

const HeadingH3 = styled.h3`
  font-size: ${font.h3};
  ${HeadingsStyle};
`

const HeadingH4 = styled.h4`
  font-size: ${font.h4};
  ${HeadingsStyle};
`

const HeadingH5 = styled.h5`
  font-size: ${font.h5};
  ${HeadingsStyle};
`

const HeadingH6 = styled.h6`
  font-size: ${font.h6};
  ${HeadingsStyle};
`

Heading.defaultProps = {
  level: "h1",
  className: null
}

Heading.propTypes = {
  children: commonPropTypes.children.isRequired,
  /**
   * The level prop determines the size of the heading, default is set to level 1 or h1
   */
  level: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  className: PropTypes.string
}
Heading.themeProps = themeProps
Heading.createVariants = componentCreateFactory(Heading)

export default Heading
