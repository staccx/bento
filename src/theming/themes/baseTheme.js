import React from "react"
import Theme from "../Theme"
import reset from "../styles/cssResets"
import grids from "../../components/Layout/Layout/Layout.Grids"
import codeTheme from "../codeTheme"

const Logo = () => <div>ThemeComponent for BaseTheme</div>
Logo.displayName = "Logo for base theme"

export const theme = {
  name: "Base Theme",
  storybookName: "Base Theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190", // for toggle eks
    link: "#279AF1"
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px",
    lineHeight: "1.5"
  },
  fontWeight: {
    bold: "bold",
    normal: "normal",
    light: "lighter"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px",
  reset,
  grid: grids,
  logo: Logo
}

export const StyleguideTheme = new Theme(theme, codeTheme)

export default theme
