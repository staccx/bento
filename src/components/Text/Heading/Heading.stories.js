import React from "react"
import Heading from "./Heading"

export default {
  title: "New/Text/Heading",
  component: Heading
}

export const Level1Header = args => (
  <Heading {...args}>This is a level 1 Heading</Heading>
)
Level1Header.args = { level: 1 }

export const Level2Header = args => (
  <Heading {...args}>This is a level 2 Heading</Heading>
)
Level2Header.args = { level: 2 }

export const Level3Header = args => (
  <Heading {...args}>This is a level 3 Heading</Heading>
)
Level3Header.args = { level: 3 }

export const Level4Header = args => (
  <Heading {...args}>This is a level 4 Heading</Heading>
)
Level4Header.args = { level: 4 }

export const Level5Header = args => (
  <Heading {...args}>This is a level 5 Heading</Heading>
)
Level5Header.args = { level: 5 }

export const Level6Header = args => (
  <Heading {...args}>This is a level 6 Heading</Heading>
)
Level6Header.args = { level: 6 }
