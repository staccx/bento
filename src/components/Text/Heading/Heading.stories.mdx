import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import Heading from "./Heading"

<Meta title="components/Text/Heading" component={Heading}/>

# Heading
Heading is Bentos header tag, it take a prop called Level with a range from 1-6.
This level tag is what sets the size of the heading.

```javascript
import { Heading } from "@staccx/bento"
```

## Standard Heading
This is a standard heading with size 1
<Preview>
  <Story name="Standard">
   <Heading level={1}>This is a Heading</Heading>
  </Story>
</Preview>


## All Sizes
This is a standard heading with size 1
<Preview>
  <Story name="All">
  <>
   <Heading level={1}>This is a level 1 Heading</Heading>
   <Heading level={2}>This is a level 2 Heading</Heading>
   <Heading level={3}>This is a level 3 Heading</Heading>
   <Heading level={4}>This is a level 4 Heading</Heading>
   <Heading level={5}>This is a level 5 Heading</Heading>
   <Heading level={6}>This is a level 6 Heading</Heading>
   </>
  </Story>
</Preview>

### Source code
<details><summary>Heading</summary>
<p>

```bash
import React from "react"
import PropTypes from "prop-types"
import styled, { css } from "styled-components"
import {
  font,
  fontFamily,
  applyVariants,
  commonPropTypes
} from "../../../theming"
import themeProps from "./Headling.themeProps"
import { componentCreateFactory } from "../../../theming/utils/createVariantsFunctionFactory"

const Heading = ({ level, children, className, ...restProps }) => {
  switch (level) {
    case "h1":
    case "H1":
    case "1":
    case 1:
      return (
        <HeadingH1 level={1} className={className} {...restProps}>
          {children}
        </HeadingH1>
      )

    case "h2":
    case "H2":
    case "2":
    case 2:
      return (
        <HeadingH2 level={2} className={className} {...restProps}>
          {children}
        </HeadingH2>
      )

    case "h3":
    case "H3":
    case "3":
    case 3:
      return (
        <HeadingH3 level={3} className={className} {...restProps}>
          {children}
        </HeadingH3>
      )

    case "h4":
    case "H4":
    case "4":
    case 4:
      return (
        <HeadingH4 level={4} className={className} {...restProps}>
          {children}
        </HeadingH4>
      )

    case "h5":
    case "H5":
    case "5":
    case 5:
      return (
        <HeadingH5 level={5} className={className} {...restProps}>
          {children}
        </HeadingH5>
      )

    case "h6":
    case "H6":
    case "6":
    case 6:
      return (
        <HeadingH6 level={6} className={className} {...restProps}>
          {children}
        </HeadingH6>
      )

    default:
      return (
        <HeadingH1 level={1} className={className} {...restProps}>
          {children}
        </HeadingH1>
      ) // ðŸ‘ˆ we always have to return something
  }
}

export const HeadingsStyle = css`
  font-family: ${fontFamily.heading};
  ${applyVariants(themeProps.heading)};
`

const HeadingH1 = styled.h1`
  font-size: ${font.h1};
  ${HeadingsStyle};
`

const HeadingH2 = styled.h2`
  font-size: ${font.h2};
  ${HeadingsStyle};
`

const HeadingH3 = styled.h3`
  font-size: ${font.h3};
  ${HeadingsStyle};
`

const HeadingH4 = styled.h4`
  font-size: ${font.h4};
  ${HeadingsStyle};
`

const HeadingH5 = styled.h5`
  font-size: ${font.h5};
  ${HeadingsStyle};
`

const HeadingH6 = styled.h6`
  font-size: ${font.h6};
  ${HeadingsStyle};
`

Heading.defaultProps = {
  level: "h1",
  className: null
}

Heading.propTypes = {
  children: commonPropTypes.children.isRequired,
  /**
   * The level prop determines the size of the heading, default is set to level 1 or h1
   */
  level: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  className: PropTypes.string
}
Heading.themeProps = themeProps
Heading.createVariants = componentCreateFactory(Heading)

export default Heading

```

</p>
</details>

## Props for Heading
<Props of={Heading} />
