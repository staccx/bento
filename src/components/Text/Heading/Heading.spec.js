import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../../theming/themes/baseTheme"
import Heading from "./Heading"

describe("Heading", () => {
  describe("Snapshots", () => {
    it("No props", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Heading>Hei</Heading>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("level 1", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Heading level={1}>Hei</Heading>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("level 2", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Heading level={2}>Hei</Heading>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("level 3", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Heading level={3}>Hei</Heading>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("level 4", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Heading level={4}>Hei</Heading>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("level 5", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Heading level={5}>Hei</Heading>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("level 6", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Heading level={6}>Hei</Heading>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Should default to h1", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Heading level={9}>Hei</Heading>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("no Children", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Heading />
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("ClassName", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Heading className="test">hei</Heading>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
