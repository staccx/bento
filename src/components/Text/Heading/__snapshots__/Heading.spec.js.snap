// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Snapshots ClassName 1`] = `
<h1
  className="sc-AxjAm gToDFS test"
>
  hei
</h1>
`;

exports[`Heading Snapshots No props 1`] = `
<h1
  className="sc-AxjAm gToDFS"
>
  Hei
</h1>
`;

exports[`Heading Snapshots Should default to h1 1`] = `
<h1
  className="sc-AxjAm gToDFS"
>
  Hei
</h1>
`;

exports[`Heading Snapshots level 1 1`] = `
<h1
  className="sc-AxjAm gToDFS"
>
  Hei
</h1>
`;

exports[`Heading Snapshots level 2 1`] = `
<h2
  className="sc-AxirZ decXcw"
>
  Hei
</h2>
`;

exports[`Heading Snapshots level 3 1`] = `
<h3
  className="sc-AxiKw cAQZNI"
>
  Hei
</h3>
`;

exports[`Heading Snapshots level 4 1`] = `
<h4
  className="sc-AxhCb cWSRul"
>
  Hei
</h4>
`;

exports[`Heading Snapshots level 5 1`] = `
<h5
  className="sc-AxhUy jDbDPK"
>
  Hei
</h5>
`;

exports[`Heading Snapshots level 6 1`] = `
<h6
  className="sc-AxgMl dSPlcW"
>
  Hei
</h6>
`;

exports[`Heading Snapshots no Children 1`] = `
<h1
  className="sc-AxjAm gToDFS"
/>
`;
